### PR TITLE
Allow access of guest accounts

### LINKED ISSUE #468: App crashes on loading if folder location cannot be accessed or created
**Description**
Loading the app fails if the recipeFolder location cannot be accessed or created.
E.g. creating the default location folder (/Recipes) is not possible for guest accounts.

**Reproduction**
1. Enable the app for guest account access
2. Login as any guest and load the app

**Expected behavior**
As the default folder location (/Recipes) cannot be accessed nor created, the app should show a file picker dialogue where the user can select the recipeFolder location to use (for a guest user, this might be a shared folder etc)

**Workaround**
manually add the recipeFolder location such that the default location is not used on app loading.
```
insert into oc_preferences (userid,appid,configkey,configvalue) values ('guest_name','cookbook','folder','Share/recipeFolderLocation');
```

**Screenshots**
Nextcloud internal server error page.

**Browser**
Firefox and Chromium

### PR DESCRIPTION
**Pull request by christianlupus**
If the user folder is not writable as typically for a guest account, the cookbook app has hight probability of being non-functional. Yet, there are no useful error messages or ways to tackle the problem. Instead a stacktrace is generated.

This should get this issue covered.

Closes #468

### CODE REVIEW

**Comment by christianlupus** [comment]
It might be good to wait for #497 to be merged as this will simplify the frontend checks probably.

**Comment by seyfeb** [comment]
I just checked the functionality (not code) and noticed the following:

- For a narrow window (either in height or width) the border is broken for menubar

Screenshot 2021-02-13 at 13 24 51

Screenshot 2021-02-13 at 13 25 09

- Selecting a shared folder as a guest doesn’t work for me
    - The selection dialog is opened and it is possible to select the folder
    - After selection nothing happens and a `500` is logged

Screenshot 2021-02-13 at 13 28 35
Screenshot 2021-02-13 at 13 31 21

File: src/components/AppInvalidGuest.vue [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+div.main {
+    height: 100%;
+    display: flex;
+    flex-direction: colummn;
+    justify-content: center;
+    align-items: center;
+
+    div.dialog {
+        width: 50%;
+        height: 40%;
```
Comment (seyfeb) [inline src/components/AppInvalidGuest.vue:71]:
```suggestion

```

File: src/components/AppInvalidGuest.vue [hunk 2]
```diff
@@ -0,0 +1,87 @@
+<template>
+    <Content app-name="cookbook">
+        <!-- <AppNavi class="app-navigation" /> -->
+        <AppContent>
+            <div class="main">
+                <div class="dialog">
+                    <div class="message">
+                        {{ t('cookbook', 'Cannot access recipe folder.') }}
+                    </div>
+                    <div class="explanation">
```
Comment (seyfeb) [inline src/components/AppInvalidGuest.vue:10]: class `explanation` is not used and may be removed

```suggestion
                    <div>
```

File: src/components/AppInvalidGuest.vue [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                        {{
+                            t('cookbook', 'You are logged in with a guest account. Therefore, you are not allowed to generate arbitrary files and folders on this nextcloud instance. To be able to use the cookbook app as a guest account, you need to specify a folder where all recipes are stored. You will need write permission to this folder.')
+                        }}
+                    </div>
+                    <div class="explanation">
```
Comment (seyfeb) [inline src/components/AppInvalidGuest.vue:15]: class `explanation` is not used and may be removed
```suggestion
                    <div>
```

File: src/components/AppInvalidGuest.vue [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        <!-- <AppNavi class="app-navigation" /> -->
+        <AppContent>
+            <div class="main">
+                <div class="dialog">
+                    <div class="message">
+                        {{ t('cookbook', 'Cannot access recipe folder.') }}
+                    </div>
+                    <div class="explanation">
+                        {{
+                            t('cookbook', 'You are logged in with a guest account. Therefore, you are not allowed to generate arbitrary files and folders on this nextcloud instance. To be able to use the cookbook app as a guest account, you need to specify a folder where all recipes are stored. You will need write permission to this folder.')
```
Comment (seyfeb) [inline src/components/AppInvalidGuest.vue:12]:
```suggestion
                            t('cookbook', 'You are logged in with a guest account. Therefore, you are not allowed to generate arbitrary files and folders on this Nextcloud instance. To be able to use the Cookbook app as a guest, you need to specify a folder where all recipes are stored. You will need write permission to this folder.')
```

File: src/components/AppInvalidGuest.vue [hunk 5]
```diff
@@ -0,0 +1,87 @@
+<template>
+    <Content app-name="cookbook">
+        <!-- <AppNavi class="app-navigation" /> -->
```
Comment (seyfeb) [inline src/components/AppInvalidGuest.vue:3]:
```suggestion
```

**Pull request merged**